lib/pages/proflile/create_usename/profile_create_username_page.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter_movie_clean/gen/colors.gen.dart';
+import 'package:flutter_movie_clean/pages/proflile/create_password/account_create_password_page.dart';
+import 'package:flutter_movie_clean/shared/extensions/context_ext.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:go_router/go_router.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+class ProfileCreateUserPage extends StatefulWidget {
+  const ProfileCreateUserPage({this.imagePath, Key? key}) : super(key: key);
+
+  static const String routeLocation = "/create_username";
+  static const String routeName = "create_username";
+
+  final String? imagePath;
+  @override
+  State<ProfileCreateUserPage> createState() => _ProfileCreateUserPageState();
+}
+
+class _ProfileCreateUserPageState extends State<ProfileCreateUserPage> {
+  final TextEditingController _usernameTextController = TextEditingController();
+  bool isKeyboardVisible = false;
+
+  bool _isEnableBtn = false;
+
+  void _updateBtnState() {
+    setState(() {
+      _isEnableBtn = _usernameTextController.text.isNotEmpty;
+    });
+  }
+
+  @override
+  void dispose() {
+    _usernameTextController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    _usernameTextController.addListener(_updateBtnState);
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    isKeyboardVisible = MediaQuery.of(context).viewInsets.bottom > 0;
+    return Scaffold(
+      backgroundColor: AppColors.black,
+      body: Column(
+        children: [
+          Expanded(
+            child: SingleChildScrollView(
+              child: Column(
+                children: [
+                  SizedBox(height: 32.h),
+                  Assets.images.icLogo.svg(),
+                  SizedBox(height: 30.h),
+                  _buildTextDescription(),
+                  SizedBox(height: 48.h),
+                  _buildAvatar(),
+                  SizedBox(height: 24.h),
+                  _buildTextFieldUsername(),
+                  SizedBox(height: 20.h)
+                ],
+              ),
+            ),
+          ),
+          _buildBtn(),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildBtn() {
+    return Container(
+      alignment: Alignment.bottomCenter,
+      margin: EdgeInsets.only(
+          bottom: isKeyboardVisible ? 20.h : 80.h, left: 30.w, right: 30.w),
+      child: SecondaryButton(
+        backgroundColor:
+            _isEnableBtn ? AppColors.crimsonApprox : AppColors.black,
+        width: 1.sw,
+        height: 50.h,
+        title: context.l10n.callMeThis,
+        onPressed: _isEnableBtn
+            ? () => context.pushNamed(
+                  AccountCreatePasswordPage.routeName,
+                  extra: {
+                    "imagePath": widget.imagePath,
+                    "username": _usernameTextController.text,
+                  },
+                )
+            : null,
+      ),
```

Comment:
i think the code looks better when we define like below
```suggestion
        onPressed: () {
             if (!_isEnableBtn) return;
             context.pushNamed(
                  AccountCreatePasswordPage.routeName,
                  extra: {
                    "imagePath": widget.imagePath,
                    "username": _usernameTextController.text,
                  },
                )
        }
      ),
```

Reply:
set `onPressed = null` in here support for no animation when user clicked button .